### ISSUE TITLE
Last line in second column of a footnote moves to next page

### ISSUE THREAD

**Issue by Alatius** [maieul/ledmac#838]
# Version

Tested both with 2018/02/28 v2.26.6 and with 2019/12/09 v2.32.3

# Description

This happens when footnotes are set in two columns: Sometimes, when a footnote is broken over two pages, the last line in the second column on the first page will be empty, and that line will be unnecessarily placed on the second page. This seems to happen when the main text is too close to the footnotes. (Change the vertical space in the MWE from 96.2 to 96.1 to see the difference.)

# Minimal Working Example

```latex
\documentclass[12pt]{article}

\usepackage[series={A},nocritical,noend]{reledmac}
\arrangementX[A]{twocol}

\begin{document}

\vspace*{96.2ex} % Test 96.1 (Ok!) and 96.2 (Error: Last line in second column moves to next page!)

Lorem.\footnoteA{Felis et semper nulla luctus ac in lacinia pharetra enim ut sodales suspendisse.}

\end{document}
```

**Comment by maieul** [comment]
It will be hard to make a general fix for that. If you get such situation concerning the position of the last line, that is because you have some trouble with your grid, especially if you have floating elements.

So, i see two solution
1) allow less space between the last line and the footnote line, using `\beforenotesX{1.2em plus 0.6em minus 0.7em}` (default value : 1em plus 0.6em minus 0.6 em)
2) solving the issue about this last line which is not in the good position.

**Comment by Alatius** [comment]
Hm, but it seems that this problem can occur whenever, even if there is no floating elements. Consider the following example:
```latex
\documentclass[12pt]{article}

\usepackage[series={A},nocritical,noend]{reledmac}
\arrangementX[A]{twocol}
\beforenotesX{1.2em plus 0.6em minus 0.7em}

\usepackage{multido}
\newcommand{\textwithfootnote}[1]{%
\multido{}{#1}{ Lorem ipsum dolor sit amet, consectetur adipiscing elite proin iaculististi.}%
\footnoteA{\multido{}{20}{ Vivamus rhoncus, urna iaculis massa.}}
\newpage~\newpage}

\begin{document}
\textwithfootnote{37}
\textwithfootnote{36}
\textwithfootnote{35}
\textwithfootnote{34}
\textwithfootnote{33}
\textwithfootnote{32}
\textwithfootnote{31}
\textwithfootnote{30}
\end{document}
```
In this example you will see several footnotes, with an increasing number of lines on their first page. Whatever number I put in \beforenotesX{}, there will be at least one footnote where the last line in the second column will be empty (on the page where the footnote starts). So it seems to be a pretty serious issue, in my opinion.

What do you mean with "trouble with your grid"?

**Comment by maieul** [comment]
by "trouble with your grid" I mean that the alignemnt of your lign in
the main text is not the same from one page to another one, and so the
height of the main text is never the multiple of the same size.

Consequently, it is difficult to know which place is available for the
footnote.

I don't say that is not a serious issue (all issues are serious).
Unfortunatly

1) I was not the original author of the two column split system
2) I have no idea of where could be the problem
3) I have no time....

**Comment by Alatius** [comment]
I understand, I know from personal experience how difficult it may be to debug things like this -- as well as to find time to spend on it! Is there anything I can do to help? This bug has been blocking a big project I am working on, and I would be ready to pay some substantial money in order to get it fixed...

**Comment by maieul** [comment]
unfortunately, problem is no money, but time... and also I have no idea why..., it is a low tech level....

**Comment by maieul** [comment]
so maybe you could find a tex expert which would have time and competencies, and I could talk with him/her about the problem...

**Comment by u-fischer** [comment]
I don't think that there is an automatic solution.  During page building TeX collects a number of footnote lines  and puts them in a box, reledmac then splits that into two parts.  The number of collected lines depends beside on the remaining place on the page and that can be so that TeX collects an odd number of lines. 

The only way I found to avoid the problem is to manually insert penalties, play around with spaces or in hard cases to manipulate the footins(A|B|C) count (everything should naturally only be done at the final stage).
